### PR TITLE
Add basic momentum optimizer

### PR DESCRIPTION
**Pull request by firasdib**
This PR adds the rudimentary momentum optimizer to the DL4J repertoar of optimizers.
The implementation is straight forward; basically a stripped down version of the nesterovs optimizer.

It is not uncommon to see these two combined into one class/function call, with a parameter defining `nesterovs = true/false`.

Either way, this seems to work for me and figured I'd share it with you.

### CODE REVIEW

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/learning/MomentumUpdater.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        if (v == null) {
+            throw new IllegalStateException("Updater has not been initialized with view state");
+        }
+
+        final double momentum = config.getMomentum();
+        final double learningRate = config.getLearningRate();
+
+        // Standard momentum: v = mu * v_prev - learning_rate * gradient(x)
+        final INDArray vPrev = v.dup(gradientReshapeOrder);
+        vPrev.muli(momentum).subi(gradient.dup(gradientReshapeOrder).muli(learningRate));
```
Comment (AlexDBlack) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/learning/MomentumUpdater.java]: This is incorrect. You dup, but never actually update the original ```v``` array - it will *always* be equal to the initial value (0), and consequently equal to vanilla SGD in practice.
Instead, it should be ```v.muli(momentum).subi(gradient.dup(order).muli(learningRate))```

File: nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/learning/MomentumUpdater.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        gradient.addi(vPrev);
```
Comment (AlexDBlack) [inline nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/learning/MomentumUpdater.java]: Should be ```gradient.assign(v)``` not addi - gradient updating is in-place, and the update is equal to ```v```

**Comment by firasdib** [comment]
@agibsonccc I'm not sure this should have been merged, its not compatible with @AlexDBlack's latest changes with regards to learning rate schedules and similar. It's a quick fix but I'm currently swamped I'm afraid.

**Comment by agibsonccc** [comment]
Will revert. From what I understood with alex, we should have only consolidated all of the learning rates to work with updaters only.

**Comment by AlexDBlack** [comment]
Yeah, just needs to be updated - PR/review was just before those changes.
